### PR TITLE
Decouple frontend from backendRoot + move plugin config into top-level Gradle module

### PR DESCRIPTION
**Pull request by PawelLipski**
This might speed up the (parallel) compilation a bit b/c now compilation of no piece of frontend is going to be blocked on compilation of any piece of backend implementation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ To run an instance of IDE with Git Machete IntelliJ Plugin execute `:runIde` Gra
 ### Generate plugin zip
 
 To generate a plugin archive run `:buildPlugin` Gradle task (`Gradle panel | Tasks | intellij | buildPlugin` or `./gradlew buildPlugin`).
-The resulting file will be available under `frontendUiRoot/build/distributions/`.
+The resulting file will be available under `build/distributions/`.
```
Comment (PawelLipski) [inline README.md:37]: I hope this is now a cleaner design (no special "designated" subproject to take the plugin artifact from).

File: frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java [hunk 2]
```diff
@@ -60,6 +64,13 @@ public GitMacheteGraphTableManager(Project project,
     subscribeToGitRepositoryChanges();
   }
 
+  @SneakyThrows
+  private static IGitMacheteRepositoryFactory getGitMacheteRepositoryFactoryInstance() {
+    Reflections reflections = new Reflections("com.virtuslab");
+    Set<Class<? extends IGitMacheteRepositoryFactory>> classes = reflections.getSubTypesOf(IGitMacheteRepositoryFactory.class);
```
Comment (PawelLipski) [inline frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java:70]: This lib looks really awesome https://github.com/ronmamo/reflections#usage

Comment (PawelLipski) [reply frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java:70]: Doing this stuff with the standard Java is PITA in comparison

File: frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java [hunk 3]
```diff
@@ -60,6 +64,13 @@ public GitMacheteGraphTableManager(Project project,
     subscribeToGitRepositoryChanges();
   }
 
+  @SneakyThrows
```
Comment (PawelLipski) [inline frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java:67]: I'm losing my patience to checked exceptions ;p https://projectlombok.org/features/SneakyThrows

Comment (PawelLipski) [reply frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java:67]: Btw later JVM languages (like Scala) ditched checked exceptions completely (in Scala, `Either` is favored for that purpose)

File: frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java [hunk 4]
```diff
@@ -60,6 +64,14 @@ public GitMacheteGraphTableManager(Project project,
...
+  private static IGitMacheteRepositoryFactory getGitMacheteRepositoryFactoryInstance() {
+    Reflections reflections = new Reflections("com.virtuslab");
```
Comment (PawelLipski) [inline frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java:69]: Big picture: we now require an `IGitMacheteRepositoryFactory` implementation to be on the **runtime classpath**, but not on the **compilation classpath** of `frontendUiTable`.

Comment (PawelLipski) [reply frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java:69]: Btw that's in essence the same trick that SLF4J is built upon.

Comment (mkondratek) [reply frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java:69]: :rocket: :exclamation:

**Pull request merged**